src/NServiceBus.Core/Routing/UnicastSubscriberTable.cs

```diff
@@ -84,9 +82,8 @@ public void AddOrReplaceRoutes(string sourceKey, IList<RouteTableEntry> entries)
             return newRouteTable.ToDictionary(kvp => kvp.Key, kvp => kvp.Value.ToArray());
         }
 
-        Dictionary<Type, UnicastRoute[]> routeTable = new Dictionary<Type, UnicastRoute[]>();
-        ConcurrentDictionary<object, IList<RouteTableEntry>> routeGroups = new ConcurrentDictionary<object, IList<RouteTableEntry>>();
-        long enterCounter;
-        long exitCounter = 1;
+        volatile Dictionary<Type, UnicastRoute[]> routeTable = new Dictionary<Type, UnicastRoute[]>();
```

Comment:
Is it? :) Why don't you need it? My reasoning was aligned to [the head in the concurrent stack](https://referencesource.microsoft.com/#mscorlib/system/Collections/Concurrent/ConcurrentStack.cs,79)